webhook/resourcesemantics/validation/validation_admit.go

```diff
@@ -158,6 +161,8 @@ func validate(ctx context.Context, resource resourcesemantics.GenericCRD, req *a
 	switch req.Operation {
 	case admissionv1beta1.Create, admissionv1beta1.Update:
 		// Supported verbs
+	case admissionv1beta1.Delete:
+		return nil // Validation handled by optional Callback, but not validatable.
```

Comment:
🤔 I don't understand the goal of excluding this.  We return `oldObj` above, so that we will get the moral equivalent of:
```go
oldObj.Validate(apis.WithinDelete(ctx))
```

This lets us add code to `Validate()` that can say:
```go
if apis.IsInDelete(ctx) {
    // Check for reasons to reject Delete
}
```

Reply:
the various validate() calls are not known to be compatible with Delete yet and might break. I would prefer to knock that out in a different PR and assess that consumers of /pkg work with that potentially unsafe change.

Reply:
This specific case was added in response to @dprotaso who worried the log line is misleading since Callbacks may be registered against delete.

Reply:
I'm fine with following up, but the point of returning `oldObj` was to make it so that `.Validate()` is a nop for implementations that haven't seen this stuff before.

It is possible that if they assume it is either create or update that the introduction of delete could still be problematic, but that seems more niche than downstream things needing to handle a nil receiver on Delete calls, which would blow up *everywhere* 😅

Reply:
that's fair... the oldObj should still be valid against, say, create.

Let's get this in and I'll patch /serving with this line knocked out (and run the tests). If everything tests fine I'll drop that line in another PR